### PR TITLE
Basic kalman filter

### PR DESCRIPTION
**Pull request by betatim**
A simple Kalman filter to estimate the position and velocity of something being dropped of a tower in Paris.

This Kalman is based on taking the mathematical expressions and directly translating them to `code`. This means it is pretty straight forward to understand. It relies on `blaze-lib` to make this fast.

Hopefully others can be persuaded to contribute other Kalman implementations which are hand tuned or explicitly use SIMD etc. As well as writing one that implements a "track fit".

### CODE REVIEW

**Comment by kdungs** [comment]
Please check if my implementation based on [Eigen](http://eigen.tuxfamily.org/) works properly and takes a sane amount of time.

Additionally, it would be nice to have a mechanism that finds libraries on the users system automatically (if present). I know CMake can do this but there might be alternatives that can be used with plain old makefiles.

**Comment by betatim** [comment]
On 4 June 2014 11:59, Kevin Dungs notifications@github.com wrote:

> Additionally, it would be nice to have a mechanism that finds libraries on
> the users system automatically (if present). I know CMake can do this but
> there might be alternatives that can be used with plain old makefiles.
> 
> I am -0 on this. As you can see a lot of code is copied and Makefiles are
> extremely simplistic. This is on purpose. I think examples which are meant
> for learning stuff need to be as explicit as possible, so that you can see
> directly in the Makefile what optimisation was used etc. Yes you could make
> it a lot nicer by abstracting that repetition away, but it would also make
> it harder to find and understand stuff and then directly edit it to find
> out what it does.

Also, your headers should be installed in a "standard" place so that you do
not need lots of commandline options to get it to compile ... this isn't
always true I know. I'd argue if it isn't true then you know what you are
doing or need to learn how to know what you are doing in order to use them
for your own projects. hence having some autoconf/cmake magic do it for you
is not in your interest. That is unless you are looking at an example of
how to perform autoconf/cmake magic.

T

**Comment by kdungs** [comment]
D’accord. From that point of view it makes perfect sense.

File: kalman/Makefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+%.o: %.cpp
+	$(CC) $(OPT) -c $<
+
+kalman: kalman.o
+	$(CC) $(LIB) -o kalman.x $^
+
+
+kalman_eigen.x: kalman_eigen.cpp
+	clang++ -O3 -std=c++11 -stdlib=libc++ -I/opt/local/include/eigen3/ -o kalman_eigen.x kalman_eigen.cpp
```
Comment (betatim) [inline kalman/Makefile]: Can we make this line look more like the blaze kalman? A la:

$(CC) $(OPT) -o kalman_eigen.x $^

and use the %.o rule to do the compiling of the .cpp file. I think adding -I$(EIGEN) to $OPT should be fine. Is that needed though if eigen is installed to a sensible (ie a default include dir) place?

**Comment by betatim** [comment]
@kdungs can you write something in the README about the eigen version and add it as a requirement

**Pull request merged**